libminifi/include/sitetosite/SiteToSiteClient.h

```diff
@@ -221,12 +221,8 @@ class SiteToSiteClient : public core::Connectable {
   // deleteTransaction
   virtual void deleteTransaction(std::string transactionID);
 
-  virtual void tearDown();
-
-  // write Request Type
-  virtual int writeRequestType(RequestType type);
-  // read Request Type
-  virtual int readRequestType(RequestType &type);
+  virtual void tearDown() = 0;
```

Comment:
i don't think this is needed at all. is it used elsewhere? Teardown is protocol specific of when and how it occurs.

Reply:
It exist in both http and raw, with different implementation, so I guess it's fine here as pure virtual.